src/HttpOverStream.NamedPipe/NamedPipeDialer.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.IO;
+using System.IO.Pipes;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace HttpOverStream.NamedPipe
+{
+    public class NamedPipeDialer : IDial
+    {
+        private readonly string _pipeName;
+        private readonly string _serverName;
+        private readonly PipeDirection _pipeDirection;
+        private readonly PipeOptions _pipeOptions;
+        private readonly int _timeout;
+
+        public NamedPipeDialer(string pipeName) 
+            : this(pipeName, ".", PipeDirection.InOut, PipeOptions.Asynchronous | PipeOptions.WriteThrough, 0)
+        {
+        }
+        public NamedPipeDialer(string pipeName, string serverName, PipeDirection pipeDirection, PipeOptions pipeOptions, int timeout)
+        {
+            _pipeName = pipeName;
+            _serverName = serverName;
+            _pipeDirection = pipeDirection;
+            _pipeOptions = pipeOptions;
+            _timeout = timeout;
+        }
+
+        public async ValueTask<Stream> DialAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            var pipe = new NamedPipeClientStream(_serverName, _pipeName, _pipeDirection, _pipeOptions);
+            await pipe.ConnectAsync(_timeout, cancellationToken).ConfigureAwait(false);
+            return pipe;
+        }
```

Comment:
I'm confused why this function doesnt use the `request` parameter. is that correct?

Reply:
Looks like the DialMessageHandler handles this request anyway so should we remove this paramter?
https://github.com/simonferquel/HttpOverStream/blob/master/src/HttpOverStream.Client/DialMessageHandler.cs#L65

Reply:
No, it gives useful information to the dialer. (in this particular cas we always call the same named pipe, but other dialers might want to extract the hostname from the request URI or use different logic for GET/POST, etc.)

Reply:
👍